src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -284,7 +285,21 @@ public void accumulate(View found) {
         }
 
         private boolean verifyLayout(Set<Integer> rules, SparseArray<View> idToChild) {
-            final TreeMap<View, List<View>> dependencyGraph = new TreeMap<View, List<View>>();
+            // We don't really care about the order, as long as it's always the same.
+            final TreeMap<View, List<View>> dependencyGraph = new TreeMap<View, List<View>>(new Comparator<View>() {
+                @Override
+                public int compare(final View lhs, final View rhs) {
```

Comment:
The treemap needs an ordering - we use it to grab the "next" view (just like your arraymap) - it turns out that Views don't have a natural ordering, so I just gave it an arbitrary one.